### ISSUE TITLE
Invalid warning/critical settings result in strange behaviour

### ISSUE THREAD

**Issue by dmzkrsk** [munin-monitoring/munin#852]
```
Linux srv1 4.4.0-79-generic #100-Ubuntu SMP Wed May 17 19:58:14 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

ii  munin                              2.0.25-2ubuntu0.16.04.3                    all          network-wide graphing framework (grapher/gatherer)
ii  munin-common                       2.0.25-2ubuntu0.16.04.3                    all          network-wide graphing framework (common)
ii  munin-doc                          2.0.25-2ubuntu0.16.04.3                    all          network-wide graphing framework (documentation)
ii  munin-node                         2.0.25-2ubuntu0.16.04.3                    all          network-wide graphing framework (node)
ii  munin-plugins-c                    0.0.9-1                                    amd64        C rewrite of some Munin plugins
ii  munin-plugins-core                 2.0.25-2ubuntu0.16.04.3                    all          network-wide graphing framework (plugins for node)
ii  munin-plugins-extra                2.0.25-2ubuntu0.16.04.3                    all          network-wide graphing framework (user contributed plugins for node)
```

I made an error in my munin config:

```
load.warning = 6.0
load.critical = 9.0
```

instead of 

```
load.load.warning = 6.0
load.load.warning = 9.0
```

and that resulted in bunch of strange `UNKNOWNs` errors appearing and disappearing at random:

```
root :: island :: pi.island :: Disk latency per device :: Average latency for /dev/sda
	UNKNOWNs: Write IO Wait time is unknown, Read IO Wait time is unknown.

root :: island :: pi.island :: Disk latency per device :: Average latency for /dev/sdb
	UNKNOWNs: Read IO Wait time is unknown, Write IO Wait time is unknown.
```

and 5 minutes later:

```
island :: pi.island :: Disk latency per device :: Average latency for /dev/sda
	OKs: Read IO Wait time is 0.00, Write IO Wait time is 0.00.

island :: pi.island :: Disk latency per device :: Average latency for /dev/sdb
	OKs: Write IO Wait time is 0.03, Read IO Wait time is 0.00.
```

and it was no only `Disk latency per device`: all other metrics disappeared at random and appeared later.

So I did some debugging and tracked the issue down to `process_limits` in `OldLimits.pm`.

I didn't found the root cause (I stopped when I found the error in my config), but that's what I have found:

* `process_limits` calls `munin_find_field_for_limits` from `Utils.pm` which walks through the whole `$config` looking for `critical|warning`.
* it found `load` key with wrongly places limits and returns it's parent node, which appears to be the root node of config (the host itself, not a specific plugin branch)
shows up for different plugins each time
* `process_service` recieves the wrong config branch and forms a wrong `$state_file` path:
   * right: `/var/lib/munin/state-island-pi.island.storable`
   * wrong: `/var/lib/munin/state-root-island.storable`
* and the final part: **for some reason all plugins after `load` plugins are also recieve the wrong root branch, try to read values from non-existent file and report `UNKNOWNs`**
  * And since `$config` `HASH` keys each time are itterated in random order, each time random plugins were affected by this error, hence, random `UNKNOWNs` in email reports.

Sadly I did not look further.

**Comment by sumpfralle** [comment]
Thank you for sharing your detailed analysis!

I could not reproduce this with v1.0.33.

Can you try it with this newer version?

**Comment by sumpfralle** [comment]
Ping?

**Comment by sumpfralle** [comment]
The issue seems to have been fixed by 35820d0dba304b9bff78f843dc30d061a213c062 (released in 2.0.26).
